### ISSUE TITLE
Reset/resend all option please...

### ISSUE THREAD

**Issue by thepurpleblob** [danmarsden/moodle-plagiarism_urkund#36]
I'm a bit vague on this but I'm hoping it makes some sense...

We noticed that if there is any kind of failure (e.g. an Urkund server outage) the submissions drop into an error state and are shown on the debug page (although we think perhaps not all are). Anyway, this is difficult to deal with as there is no 'resend all' option.

More fundamentally, we are wondering why the plugin doesn't just retry with an increasing backoff time (same sort of idea as SMTP implementations) when there is a failure?

**Comment by nomadwarmachine** [comment]
Just to add to this: on Sunday night/Monday morning there was an Urkund server outage, meaning that when I checked on Monday morning the submissions during that time were sitting within Moodle Assignment with the error icon and a "reset" option.  None of these were showing on the debug page, I had to check each Assignment individually to reset them - so if I had not been monitoring for our pilot this would not have been picked up until academics or students raised it with our Help Desk.

In addition to the above, I'd hope that the debug page could have options to "clear all" as well as "resend all".

**Comment by danmarsden** [comment]
Thanks Howard - I think I intended to add a retry process for failed submissions but it looks like I only implemented the retry-handling to the check to obtain the similarity score. I'll take a look to see what would be needed to add this in. I did implement this in the old Turnitin plugin but was never happy with the implementation as it held up the events queue. We should be able to do this much easier with urkund.
